pkg/collector/container_hc_collector.go

```diff
@@ -84,14 +92,20 @@ func (c *Collector) updateHWCounters(containerID string, ct *ProcessBPFMetrics,
 				klog.V(5).Infoln(err)
 			}
 		}
+		// track virtual machine metrics
+		if isSystemVM && config.EnableProcessMetrics {
+			c.VMMetrics[ct.PID].CounterStats = c.ProcessMetrics[ct.PID].CounterStats
+		}
 	}
 }
 
 // updateBPFMetrics reads the BPF tables with process/pid/cgroupid metrics (CPU time, available HW counters)
 func (c *Collector) updateBPFMetrics() {
 	foundContainer := make(map[string]bool)
 	foundProcess := make(map[uint64]bool)
+	foundVM := make(map[uint64]bool)
 	processesData, err := attacher.CollectProcesses()
+	vmPIDList, _ := libvirt.GetCurrentVMPID()
```

Comment:
@aharivel @rootfs We should retrieve the VM ID in the same way we do for the container ID. The libvirt ID can be found in `/proc/PID/cgroup`. This approach will allow us to maintain consistent and efficient code, eliminating the need to iterate through the vmPIDList for all ebpf entries.